src/search/SearchMessagesCard.js

```diff
@@ -18,23 +18,26 @@ const styles = StyleSheet.create({
 });
 
 type Props = {|
-  queryIsEmpty: boolean,
-  messages: Message[],
+  messages: Message[] | null,
   isFetching: boolean,
 |};
 
 export default class SearchMessagesCard extends PureComponent<Props> {
   static NOT_FETCHING = { older: false, newer: false };
 
   render() {
-    const { queryIsEmpty, isFetching, messages } = this.props;
+    const { isFetching, messages } = this.props;
 
     if (isFetching) {
       return <LoadingIndicator size={40} />;
     }
 
+    if (messages === null) {
+      return null;
+    }
+
     if (messages.length === 0) {
```

Comment:
Small style bit: perhaps make these three conditionals an "if / else if / else if" chain? Saves a few lines of vertical space, and semantically they're quite closely related so I think it's helpful in that respect too.

In the later commit that adds a condition inside `isFetching`, that'll mean the extra condition needs to be an `&&` conjunct instead of a nested `if`; I think that's a (nit-scale) improvement too.